app/controllers/portal/client_logins_controller.rb

```diff
@@ -44,10 +44,14 @@ def check_verification_code
         if Rails.configuration.allow_magic_verification_code && @verification_code_form.verification_code == "000000"
           update_existing_token_with_magic_code(hashed_verification_code)
         end
-        @records = client_login_service.login_records_for_token(hashed_verification_code)
-        return if redirect_locked_clients # check if any records are already locked
-        if @records.present? # we have at least one match and none are locked
+
+        verified_matching_records = client_login_service.login_records_for_token(hashed_verification_code)
+        @records = verified_matching_records.present? ? verified_matching_records : failed_verification_matching_records
+        unlock_matching_records
```

Comment:
i ended up refactoring b/c i really like knowing what our `@records` are

Reply:
this is very clean in a very strange flow! thanks arin :)